cli/deno_dir.rs

```diff
@@ -265,95 +296,74 @@ impl DenoDir {
-    let referrer = self.src_file_to_url(referrer);
-
-    debug!(
-      "resolve_module specifier {} referrer {}",
-      specifier, referrer
-    );
+    url: &Url,
+  ) -> Result<PathBuf, DenoDirError> {
+    let filename = match url.scheme() {
+      "file" => url.to_file_path().unwrap(),
+      "https" => get_cache_filename(self.deps_https.as_path(), &url),
+      "http" => get_cache_filename(self.deps_http.as_path(), &url),
+      scheme => {
+        return Err(
+          DenoDirError::new(
+            format!("Unsupported scheme \"{}\" for module \"{}\". Supported schemes: {:#?}", scheme, url, SUPPORTED_URL_SCHEMES),
+            DenoDirErrorKind::UnsupportedFetchScheme
+          )
+        );
+      }
+    };
 
-    resolve_file_url(specifier, referrer)
+    debug!("deps filename: {:?}", filename);
+    Ok(normalize_path(&filename))
   }
 
-  // TODO(bartlomieju): this method should return only `local filepath`
-  //  it should be called with already resolved URLs
-  // TODO(bartlomieju): rename to url_to_deps_path
-  /// Returns (module name, local filename)
-  pub fn resolve_module(
-    self: &Self,
-    specifier: &str,
-    referrer: &str,
-  ) -> Result<(String, PathBuf), url::ParseError> {
-    let j = self.resolve_module_url(specifier, referrer)?;
-
-    let module_name = j.to_string();
-    let filename = match j.scheme() {
-      "file" => j.to_file_path().unwrap(),
-      "https" => get_cache_filename(self.deps_https.as_path(), &j),
-      "http" => get_cache_filename(self.deps_http.as_path(), &j),
-      // TODO(kevinkassimo): change this to support other protocols than http.
-      _ => unimplemented!(),
-    };
+  // TODO: this method is only used by `SourceMapGetter` impl - can we organize it better?
+  fn try_resolve_and_get_module_meta_data(
+    &self,
+    script_name: &str,
+  ) -> Option<ModuleMetaData> {
+    // if `script_name` can't be resolved to ModuleSpecifier it's probably internal
+    // script (like `gen/cli/bundle/compiler.js`) so we won't be
+    // able to get source for it anyway
+    let maybe_specifier = ModuleSpecifier::resolve_absolute(script_name);
 
-    debug!("module_name: {}, filename: {:?}", module_name, filename);
-    Ok((module_name, normalize_path(&filename)))
+    if maybe_specifier.is_err() {
+      return None;
+    }
+
+    let module_specifier = maybe_specifier.unwrap();
+    // TODO: this method shouldn't issue `fetch_module_meta_data` - this is done for each line
+    //  in JS stack trace so it's pretty slow - quick idea: store `ModuleMetaData` in one
+    //  structure available to DenoDir so it's not fetched from disk everytime it's needed
+    match self.fetch_module_meta_data(&module_specifier, true, true) {
+      Err(_) => None,
+      Ok(out) => Some(out),
+    }
   }
 }
 
 impl SourceMapGetter for DenoDir {
   fn get_source_map(&self, script_name: &str) -> Option<Vec<u8>> {
-    match self.fetch_module_meta_data(script_name, true, true) {
```

Comment:
It's not so important but you can rewrite this entire function as one line:
`self.fetch_module_meta_data(script_name, true, true).and_then(|out| out.maybe_source_map)`

Reply:
Thanks!